src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -878,22 +849,18 @@ protected virtual IDbDataParameter CreateDatabaseParameter(IDbCommand command, D
                 dbParameter.Scale = parameterInfo.Scale;
             }
 
-            _parameterTypeSetter.SetParameterDbType(dbParameter, parameterInfo);
-            object value;
-
             try
             {
-                value = GetParameterValue(logEvent, parameterInfo, dbParameter.ParameterName, dbParameter.DbType);
+                if (!parameterInfo.SetDbType(dbParameter))
+                {
+                    InternalLogger.Warn("  DatabaseTarget: Parameter: '{0}' - Failed to assign DbType={1}", dbParameter.ParameterName, parameterInfo.DbType);
```

Comment:
maybe another message than the one below? That is easier to recognize the issue. Also I think both are the same level (error or warning) - the have the same result

Reply:
One complains about unknown DbType. The other says something horrible went wrong when trying to deduce and assign the DbType.